BLSQ/iaso
# POLIO-1645 POLIO-1646  POLIO-1647  incident reports improvements

Explain what problem this PR is resolving

Related JIRA tickets : POLIO-1645 POLIO-1646  POLIO-1647  

## Self proofreading checklist


## Doc
Some doc is present on plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Modals/CreateEditIncident.tsx

## Changes

- added title and comment to incident report model + migrations
- added those field in form
- added Other option in types, not displayed in the frontend for now as POLIO-1650 ticket is hold due to the complexity of the feature
- review number inputs logic on the frontend to make it clear and match to this:
<img width="845" alt="Screenshot 2024-08-22 at 12 23 26" src="https://github.com/user-attachments/assets/664d6ca7-02e0-4a48-bd65-7c6a92a4a82c">

## How to test

Open a stock variation for a country with existing data. 
<img width="350" alt="Screenshot 2024-08-22 at 16 03 04" src="https://github.com/user-attachments/assets/fb379ed0-2a2a-4757-aa9e-cfd638030885">

open the details for one entry and click on stock variation.
Select Incident reports tab .
Edit and create reports, check it matches the sheet
Add title and comment and save.

## Print screen / video

<img width="809" alt="Screenshot 2024-08-22 at 15 58 01" src="https://github.com/user-attachments/assets/6e6346f5-af69-47df-8bee-da22ef7fbb66">
<img width="977" alt="Screenshot 2024-08-22 at 15 57 55" src="https://github.com/user-attachments/assets/b1643050-134a-415c-be56-36b5a7cc9581">
<img width="742" alt="Screenshot 2024-08-22 at 15 57 49" src="https://github.com/user-attachments/assets/16b2769e-52f8-40e2-833a-419735c53978">
<img width="1646" alt="Screenshot 2024-08-22 at 15 57 43" src="https://github.com/user-attachments/assets/e7a21147-5d44-45ec-ae49-81f81aac36ef">
